Make _compute_linear_combination.out a true out function

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43272 Make _compute_linear_combination.out a true out function**

Was missing kwarg-onlyness.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D23215506](https://our.internmc.facebook.com/intern/diff/D23215506)

ezyang committed:
Make _compute_linear_combination.out a true out function

Was missing kwarg-onlyness.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

nikitaved commented on aten/src/ATen/native/FunctionOfAMatrixUtils.cpp:
```diff
@@ -24,15 +24,15 @@ Tensor _compute_linear_combination(const Tensor& input, const Tensor& coefficien
 // which means that if `output` actually is aliased by `input`, the result
 // produced is undefined.
-Tensor _compute_linear_combination_out(const Tensor& input, const Tensor& coefficients, Tensor& output) {
+Tensor& _compute_linear_combination_out(Tensor& output, const Tensor& input, const Tensor& coefficients) {
```
Yeah, I was curious why other `out` methods had `output` as the very first argument. Looks like I completely missed  the schema. 

Actually, the `out` is not being used in `matrix_exp`, but it is good to have. I will probably add it to the existing code to have one larger contiguous buffer instead of separate two.

nikitaved reviewed (approved):
Thank you!

ezyang committed:
Update on "Make _compute_linear_combination.out a true out function"


Was missing kwarg-onlyness.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D23215506](https://our.internmc.facebook.com/intern/diff/D23215506)

[ghstack-poisoned]
